pkg/parts/parts.go

```diff
@@ -109,9 +167,39 @@ type Part struct {
 	Sheets Links `json:"sheet,omitempty"`
 }
 
+func (x *Part) RedisKey() string {
+	return fmt.Sprintf("%s:%s:%d", x.ID.Project, x.ID.Name, x.ID.Number)
+}
+
+func (x *Part) DecodeRedisKey(str string) {
+	got := strings.SplitN(str, ":", 3)
+	x.ID.Project = got[0]
+	x.ID.Name = got[1]
+	num, _ := strconv.Atoi(got[2])
+	x.ID.Number = uint8(num)
+}
+
+func (x *Part) ZScore() int {
+	projectScore, _ := strconv.Atoi(x.ID.Project[:2])
+	return projectScore
+}
+
 type Link struct {
-	ObjectKey string
-	CreatedAt time.Time
+	ObjectKey string    `json:"object_key"`
+	CreatedAt time.Time `json:"created_at"`
+}
+
+func (x *Link) ZScore() int {
+	return int(x.CreatedAt.Unix())
+}
+
+func (x *Link) EncodeRedisString() string {
+	linkBytes, _ := json.Marshal(x)
+	return string(linkBytes)
+}
+
+func (x *Link) DecodeRedisString(src string) {
+	json.Unmarshal([]byte(src), x)
 }
```

Comment:
Is `EncodeX`/`DecodeX` or `ToX`/`FromX` more idiomatic? In Rust, I believe they each have specific meanings, but IDK about Go.

Reply:
I'm not sure what you mean.